doc/rst/tools/mason/mason.rst

```diff
@@ -0,0 +1,339 @@
+
+.. _readme-mason:
+
+=====
+mason
+=====
+
+Mason represents an effort by the developers on the chapel team at Cray to reach
+out to the chapel community and provide a systematic way of providing packages
+to the developers who want to build and explore in chapel. 
+
+
+
+
+
+Installation Instructions
+=========================
+
+In the mason directory found in ``$CHPL_HOME/tools/mason`` run the following:
+
+.. code-block:: sh
+   
+   make mason
+
+
+This will create the hidden ``.mason/`` directory in ``MASON_HOME``, which
+defaults to your ``$HOME`` if unset. It builds the mason binary so that the 
+command line interface can be used. The mason registry is also downloaded from
+github so that a user can start to rely on mason to specifcy project
+dependencies. 
+
+.. code-block:: sh
+
+   make install
+
+This puts mason in the path of the user so that mason can be used
+anywhere in the user's file system.
+
+
+To do this all in one step, type:
+
+.. code-block:: sh
+
+   make all
+
+
+
+To remove mason simply change directory to ``$CHPL_HOME/tools/mason`` and run:
+
+.. code-block:: sh
+
+   make clean
+      
+
+
+
+
+
+Setting up your Project
+=======================
+	
+``mason new [project name] [--vcs=true/false]`` is the command that initializes
+a new project. It also creates a git repository unless ``--vcs=false`` is included.
+
+For example, after ``mason new MyPackage`` is run in an empty directory it will have the 
+following hierarchy::
+
+	MyPackage/
+  	  Mason.toml
+  	  src/
+    	    MyPackage.chpl
+
+Mason will enforce that the main file be named after the package to enforce namespacing.
+While it is common practice for package names to be PascalCase and chpl files to be lowercase,
+it is an acceptable tradeoff for reliability, and MyPackage will be the first file listed in ``src/``.
+
+
+
+
+
+
+Building and Running your project
+=================================
+
+When invoked, ``mason build`` will do the following:
```

Comment:
We should mention how that `mason` forwards compile/runtime flags somewhere around here...

Also, I discovered an ugly caveat: 

Because `mason` is a Chapel program, it consumes all the default flags like `-v`, `-s`, `-nl`... This is annoying and problematic. Long-term, we'd like to create a way to disable these default flags for a Chapel program to disable this. In the near-term, this can be overridden by passing flags after a `--`.

So to build a Chapel program with a param flag via `mason`, you'd need to do:

```
> mason build -- -smyParamFlag=true
```

whereas the following will not work:

```
> mason build -smyParamFlag=true
<command-line arg>:2: error: Unrecognized configuration variable 'myParamFlag'
```